File: leetcode.py/合并数组.py
class Solution(object):
    def merge(self, intervals):
        if not intervals:
            return []

        intervals.sort()

        merged = [intervals[0]]
        for current in intervals[1:]:
            last = merged[-1]
            if current[0] <= last[1]:
                merged[-1][1] = max(current[1] ,last[1])
            else:
                merged.append(current)
        return merged
        # 示例用法
if __name__ == "__main__":
    solution = Solution()
    intervals = [[1, 3], [2, 6], [8, 10], [15, 18]]
    result = solution.merge(intervals)
    print(f"合并后的区间: {result}")  # 预期输出: [[1, 6], [8, 10], [15, 18]]

File: leetcode.js/合并数组.js
var merge = function(intervals) {
    if(!intervals){
        return [];
    }
    intervals.sort((a,b) => a[0] - b [0]);
    let merged = [intervals[0]];

    for(let i = 1;i < intervals.length ; i++){
        const lastMerged = merged[merged.length - 1];
        const current = intervals[i];
        if (lastMerged[1] >= current[0]){
            lastMerged[1] = Math.max(lastMerged[1],current[1]);
        }
        else{
            merged.push(current);
        }
    }
    return merged;
};
// 示例用法
const intervals = [[1,3],[2,6],[8,10],[15,18]];
const result = merge(intervals);
console.log(result); // 输出: [[1,6],[8,10],[15,18]]
